### ISSUE TITLE
508-defect-2 [COGNITION]: How to apply for healthcare Spanish content must have a proper LANG attribute

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-cms#13148]
## Details

There is an additional information component on the register for health care page that needs a `lang="es"` attribute. Without this attribute, screen readers attempt to read Spanish content with the English pronunciation dictionary. Context and intent are reduced or lost this way. By adding the proper attribute, screen readers will load and use the Spanish dictionary. Screenshot attached below.

## Engineering notes

The widget is not always translated. So the lang attribute needs to be added dynamically based on context.

## Acceptance Criteria

- [ ] The appropriate (Spanish, Tagalog) lang attribute is added to content container 
- [ ] Screen reader uses the correct inflection and pronunciation. This can be tested by listening to content with a screen reader before the lang attribute is applied, and then again after. There should be a different intonation in the voice announcement, and with correct Spanish pronunciations.

## Environment

* https://va.gov/health-care/how-to-apply/
* CMS node: https://prod.cms.va.gov/node/35/edit

## Proposed Solution (if known)

```diff
<div
  data-template="paragraphs/spanish_translation"
  data-entity-id="3423"
  class="form-expanding-group additional-info-container form-expanding-group-open"
+ lang="es"
>...</div>
```

## WCAG or Vendor Guidance (optional)

* [Language of Parts: Understanding SC 3.1.2](https://www.w3.org/TR/UNDERSTANDING-WCAG20/meaning-other-lang-id.html)

## Screenshots or Trace Logs

![Screen Shot 2021-03-05 at 4 31 45 PM](https://user-images.githubusercontent.com/934879/110181345-2fe36280-7dd1-11eb-9ce8-47d6955bfba4.png)

**Comment by jilladams** [comment]
@laflannery can you confirm that you listened to this page via screenreader and heard the Spanish start / stop? 

From planning: 
https://www.va.gov/health-care/how-to-apply-esp/

- "en" is declared in page <html> element
- `lang="es"` is declared at the <article> level
- Language is not declared on expanding groups. This is where investigation should start.

**Comment by wesrowe** [comment]
@laflannery, the team (and Dave C) didn't feel they could estimate this without knowing more about how widespread the issue is. It would be helpful if you could shed light on that – is it a problem with how additional-info components are templated, in which case it would affect multiple pages?

**Comment by laflannery** [comment]
@jilladams I confirmed that on the Spanish page you sent me the screen reader (VoiceOver) does switch languages correctly based on the `lang` attribute that is set for the content that is being read. I believe that is what is being asked.

> Language is not declared on expanding groups. This is where investigation should start.

I'm not sure I understand this and what is being asked here. On the page you sent me the lang does not need to be set because the language has not changed from what was set as the main article language. For the page in the description the reason it needs to be set is because the language is different from what the main content language is. Let me know if we should discuss further.

@wesrowe I was only made aware of this because I found it in the backlog. I can dig further an confirm that this is an issue for any additional info component but my guess is yes because I do not believe there is a way for folks to set the language for a particular part of a page.

**Comment by laflannery** [comment]
@wesrowe Confirming my thought above (I used staging) - 

I have no way, as an editor to indicate that I have changed languages from the main content language so I could not indicate that my additional information is actually in Spanish:

Screenshot from the front end:
There is no `lang` attribute specified so the main content language (in this case EN) will be assumed
![image](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/106678594/6735ecfb-5597-47f3-a465-b111361d081c)

Screenshot from the CMS:
I only have the text fields, no language indicator field
![image](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/106678594/7037f431-423a-4f49-98d2-e9dc64fe35ce)

**Comment by wesrowe** [comment]
@laflannery, can you clarify, if you know, why this particular additional-info component needs its own `lang` attribute instead of inheriting from some higher level of the DOM?

**Comment by laflannery** [comment]
@wesrowe see my comment above to Jill but it’s because the language has changed. Suddenly, in a page full of English, I’m adding Spanish content so there needs to be a way to indicate this change. Let me know if that is clear or if we should get on a call for me to explain further.

**Comment by randimays** [comment]
It's occurred to me that we don't seem to have a logical place to determine the language from the front end. There is no language selection on the CMS side when this text is entered and the template is used specifically when English is the language for the whole page. Asking the team for more clarification about what was the original plan for resolving this from the front end only.

**Comment by jilladams** [comment]
Post scrum: https://dsva.slack.com/archives/C52CL1PKQ/p1684858075198999
* Daniel verified that the paragraph type in CMS was previously named "Spanish translation" and renamed to "Additional information" in 3/2019. Ticket referred to is in an old repo we don't have access to afaik (VAGOV-1406)
* 124 instances of the paragraph, most in English (based on skim of heading / label text). TBD how we can review those in more depth.
* In order to fix the FE a11y issue, we'll have to provide a means for the CMS to tell the FE when the content is in Spanish.

This ticket is likely blocked til we work that out, as we can't modify the template without affecting the English content.

**Comment by wesrowe** [comment]
@DanielleThierryUSDSVA, This 508/a11y ticket was "caused" by the use of a Spanish-language "additional info" component in the middle of an English-language page. (Screen readers were not switching pronunciation/accent.) I realized that before we re-engineer to make this possible, we should ask if you feel this language-switch capability is important. Thoughts?

Links to example/instance:
* https://va.gov/health-care/how-to-apply/
* CMS node: https://prod.cms.va.gov/node/35/edit

**Comment by DanielleThierryUSDSVA** [comment]
Hi @wesrowe. I'm so glad you asked! This approach was a super old Band Aid for providing some level of Spanish language access for the health care application. Now that we have the full page translation available with the Spanish toggle, we really don't even need this anymore. I'll confirm with Randi that we're in the process of removing this from the page.

**Comment by wesrowe** [comment]
Thanks, @DanielleThierryUSDSVA and @rlhecht! Can I ask you to update this ticket when you've removed the content? 

@laflannery, FYSA.

**Comment by jilladams** [comment]
Moving out of sprint and we will wait for content team updates before closing.

**Comment by laflannery** [comment]
I'm going to go ahead and close, thank you everyone (@jilladams @wesrowe @RLHecht) for working to get this sorted and resolved, Much appreciated!!
